folding-schemes/src/folding/nova/mod.rs

```diff
@@ -392,21 +377,11 @@ where
     CS2: CommitmentScheme<C2, H>,
 {
     fn deserialize_with_mode<R: std::io::prelude::Read>(
-        mut reader: R,
-        compress: ark_serialize::Compress,
-        validate: ark_serialize::Validate,
-    ) -> Result<Self, ark_serialize::SerializationError> {
-        let r1cs = R1CS::deserialize_with_mode(&mut reader, compress, validate)?;
-        let cf_r1cs = R1CS::deserialize_with_mode(&mut reader, compress, validate)?;
-        let cs_vp = CS1::VerifierParams::deserialize_with_mode(&mut reader, compress, validate)?;
-        let cf_cs_vp = CS2::VerifierParams::deserialize_with_mode(&mut reader, compress, validate)?;
-        Ok(VerifierParams {
-            poseidon_config: poseidon_canonical_config::<C1::ScalarField>(),
-            r1cs,
-            cf_r1cs,
-            cs_vp,
-            cf_cs_vp,
-        })
+        mut _reader: R,
+        _compress: ark_serialize::Compress,
+        _validate: ark_serialize::Validate,
+    ) -> Result<Self, SerializationError> {
+        unimplemented!("use FoldingScheme::vp_deserialize_with_mode method instead");
```

Comment:
Should we not implement this trait at all eventually? This might help users avoid making this mistake.

Reply:
The thing is that previously it was serializing the whole R1CS of the main circuit and the R1CS of the CycleFold circuit, which was taking several tens/hundreds of MBs.
The new approach recomputes both R1CS on the fly, so that they don't need to be serialized, saving up tens/hundreds of MBs depending on the folding circuit. To do so, it needs the FCircuit but also it's initializing parameters, and also the GC1 and GC2, in order to rebuild it, but the GC1 & GC2 (CurveVar) don't allow for a serialization (as you also faced and mentioned in the PR https://github.com/privacy-scaling-explorations/sonobe/pull/107 ). So the way around has been to do the deserialization logic from the Nova struct itself, which can easily reconstruct the circuits from the given trait elements and thus recompute the two R1CS to rebuild the VerifierParams.

Reply:
Makes sense, but it seems like this implementation can be removed without taking out any functionality, if not making users less confused?

Reply:
Yes, but thought about giving a hit to the dev that uses the lib and expects to call the deserialize method, so that they know that it is not supported (hence the `unimplemented` message redirecting to the other method).

Reply:
I understand, maybe showing devs how to serde could be documented instead of doing it with additional LOCs?